lib/biz/time.rb

```diff
@@ -39,7 +40,12 @@ def on_date(date, day_time)
         true
       )
     rescue TZInfo::PeriodNotFound
-      on_date(date, DayTime.new(day_time.day_second + HOUR))
+      next_day_second = day_time.day_second + HOUR
+      if next_day_second > SECONDS_IN_DAY
+        on_date(date + 1, DayTime.new(next_day_second % SECONDS_IN_DAY))
```

Comment:
Do we know it's going to be 86400 + 3600 here?  Can just say `DayTime.new(HOUR)` here?